cmd/export_ledgers.go

```diff
@@ -54,13 +82,56 @@ func printTransformStats(attempts, failures int, printLog bool) {
 
 	results, err := json.Marshal(resultsMap)
 	if err != nil {
-		cmdLogger.Fatal("Could not marshall results: ", err)
+		cmdLogger.Fatal("Could not marshal results: ", err)
+	}
+
+	cmdLogger.Info(string(results))
+}
+
+func generateRunId() string {
+	return uuid.New().String()
+}
+
+func uploadToGcs(credentialsPath, bucket, runId, path string) error {
+	// Use credentials file in dev/local runs. Otherwise, derive credentials from the service account.
+	if len(credentialsPath) > 0 {
+		os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", credentialsPath)
+		cmdLogger.Infof("Using credentials found at: %s", credentialsPath)
+	}
+
+	reader, err := os.Open(path)
+	if err != nil {
+		return fmt.Errorf("failed to open file %s: %v", path, err)
+	}
+
+	ctx := context.Background()
+	client, err := storage.NewClient(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to create client: %v", err)
+	}
+	defer client.Close()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*60)
+	defer cancel()
+
+	objectPath := filepath.Join(runId, path)
+	wc := client.Bucket(bucket).Object(objectPath).NewWriter(ctx)
+	defer wc.Close()
```

Comment:
Would it be useful to check the `err` in the `.Close()` method? Are there cases in GCS where the `.Close()` function would error and we wouldn't know? (Like timeouts or incomplete file writes?)

Reply:
Agreed that it would be helpful to surface that error. Added error checking for it.